### PR TITLE
feature: recognize domain specific words

### PR DESCRIPTION
**Pull request by densumesh**

### CODE REVIEW

**Review by skeptrunedev — COMMENTED** [review]
Are the dependency removals here intentional?

**Comment by densumesh** [comment]
> Are the dependency removals here intentional? 

Yes they are. No longer using bloom filters and bktree functions have been moved into our code

File: server/src/operators/typo_operator.rs [hunk 1]
```diff
@@ -650,13 +660,20 @@ fn correct_query_helper(tree: &BkTree, query: String, options: &TypoOptions) ->
...
 
     if corrections.is_empty() {
-        None
+        CorrectedQuery {
+            query: Some(query),
+            corrected: false,
+        }
     } else {
-        let mut corrected_query = query.to_string();
+        let mut corrected_query = query.query.clone();
```
Comment (skeptrunedev) [inline server/src/operators/typo_operator.rs:668]: Behavior is correct, but it shouldn't always happen. `TypoOptions` needs another param, which defaults to True, such that the user can turn this behavior off if they want to.

**Review by skeptrunedev — CHANGES_REQUESTED** [review]

**Review by skeptrunedev — CHANGES_REQUESTED** [review]
new request param needs to get added to options popover

**Review by skeptrunedev — APPROVED** [review]

**Pull request merged**